web/public/js/directives/cd-child-card/index.js

```diff
@@ -0,0 +1,192 @@
+              originalDojoListings = undefined;
+              originalDojosNotAMemberOf = undefined;
+              ctrl.manageDojos = false;
+              ctrl.savingDojos = false;
+            }
+          })
+        });
+      };
+
+      ctrl.addToDojo = function (e, dojoToAdd) {
+        e.preventDefault();
+        var addedDojo = _.remove(ctrl.child.dojosNotMemberOf, function (dojo) {
+          return dojo.id === dojoToAdd.id;
+        });
+        ctrl.child.dojoListings = _.concat(ctrl.child.dojoListings, addedDojo);
+        ctrl.sortDojos();
+      };
+
+      ctrl.removeFromDojo = function(e, dojoToRemove) {
+        e.preventDefault();
+        var removedDojo = _.remove(ctrl.child.dojoListings, function (dojo) {
+          return dojo.id === dojoToRemove.id;
+        });
+        ctrl.child.dojosNotMemberOf = _.concat(ctrl.child.dojosNotMemberOf, removedDojo);
+        ctrl.sortDojos();
+      }
+
+      ctrl.sortDojos = function () {
+        ctrl.child.dojosNotMemberOf = _.sortBy(ctrl.child.dojosNotMemberOf, ['caption']);
+        ctrl.child.dojoListings = _.sortBy(ctrl.child.dojoListings, ['caption']);
+      };
+
+      ctrl.prepareDojoListings = function () {
+        var child = ctrl.child;
+        child.dojoIds = [];
+        child.dojoListings = child.dojos.map(function (dojo) {
+          if (child.dojoIds.indexOf(dojo.id) === -1) {
+            child.dojoIds.push(dojo.id);
+          }
+          var pictureGridItem = {
+            id: dojo.id,
+            href: 'dojo/' + dojo.urlSlug,
+            picture: '/img/avatars/dojo-default-logo.png',
+            caption: dojo.name
+          };
+          cdDojoService.getAvatar(dojo.id)
+            .then(function (avatarUrl) {
+              pictureGridItem.picture = avatarUrl;
+            });
+          return pictureGridItem;
+        });
+        child.dojosNotMemberOf = [];
+        child.dojosNotMemberOf = ctrl.parentProfileData.dojos.filter(function (dojo) {
+          return child.dojoIds.indexOf(dojo.id) === -1;
+        }).map(function (dojo) {
+          var pictureGridItem = {
+            id: dojo.id,
+            href: 'dojo/' + dojo.urlSlug,
+            picture: '/img/avatars/dojo-default-logo.png',
+            caption: dojo.name
+          };
+          cdDojoService.getAvatar(dojo.id)
+            .then(function (avatarUrl) {
+              pictureGridItem.picture = avatarUrl;
+            });
+          return pictureGridItem;
+        });
+        ctrl.sortDojos();
+      };
+
+      ctrl.prepareBadgeListings = function () {
+        var child = ctrl.child;
+        if (child.badges) {
+          child.acceptedBadges = child.badges.filter(function (badge) {
+            return badge.status === 'accepted';
+          });
+        } else {
+          child.acceptedBadges = [];
+        }
+        child.recentBadges = child.acceptedBadges.sort(function (badge1, badge2) {
+          var d1 = new Date(badge1.dateAccepted);
```

Comment:
sort again, do we need another alg?
Or export to utils since it's the "same" than before